### PR TITLE
BDOG-21 Catalogue doesn't understand -play-25 versions

### PR DESCRIPTION
**Pull request by christopherjturner**

### CODE REVIEW

**Comment by konradwudkowski** [comment]
I'm not sure if this is part of `services-dependencies` or rather `catalogue-frontend` but do we want to show play-26 dependencies on a play 2.6 microservice and play-25 ones on play 2.5 microservices? Only returning major,minor,patch may not enable this.

File: test/uk/gov/hmrc/servicedependencies/model/VersionParserSpec.scala [hunk 1]
```diff
@@ -139,6 +139,29 @@ class VersionParserSpec extends FreeSpec with MustMatchers {
...
 
+  "Parsing version ending with play version returns correct verson" in {
+    val buildFile = """  object Test {
+                      |    def apply() = new TestDependencies {
+                      |      override lazy val test = Seq(
+                      |        "uk.gov.hmrc" %% "simple-reactivemongo" % "7.0.0-play-26",
+                      |      )
+                      |    }.test
+                      |  }""".stripMargin
+  }
```
[the remark this answers is not in the document]
Comment (christopherjturner) [reply test/uk/gov/hmrc/servicedependencies/model/VersionParserSpec.scala]: Yes, looks like there should be

**Comment by christopherjturner** [comment]
Sounds like returning the version's suffix would be the right thing to do. Doesn't really make sense for the service to be discarding this data.
I'll look at adding it in.

**Comment by konradwudkowski** [comment]
We could also return `major`,`minor`,`patch` and one more field e.g. `playCrossCompiled = true/false`. Then the Catalogue could infer the version of play from a plugin that is used and add a correct suffix of `-play-25` or `-play-26`. Just a thought, maybe not a good one.

**Comment by christopherjturner** [comment]
Not sure how easily we can identify if a given library version is cross compiled. 
It looks like the service finds latest/alternative library versions by looking at the libraries repo's tags, which only have the version (e.g. v1.2.3). Might be possible to parse it from the release name or look it up in the artefact repository?

**Comment by christopherjturner** [comment]
I've raised a new ticket (BDOG-44) to implement this.

**Pull request merged**